tools/benchmark.cr

```diff
@@ -0,0 +1,57 @@
+require "../src/pluto"
+
+def benchmark_memory
+  bytes_before_measure = GC.stats.total_bytes
+  yield
+  (GC.stats.total_bytes - bytes_before_measure).to_i64
+end
+
+def benchmark_time
+  time = Time.monotonic
+  yield
+  Time.monotonic - time
+end
+
+def output_results(total : Hash(String, NamedTuple(memory: Int64, time: Int64)))
+  name_rjust = total.keys.map(&.size).max
+  time_ljust = total.values.map { |t| t[:time].to_s.size + 2 }.max
+  mem_rjust = total.values.map(&.[:memory].humanize_bytes.size).max
+  puts "| #{"".ljust(name_rjust)} | #{"Time".ljust(time_ljust)} | #{"Memory".ljust(mem_rjust)} |"
+  puts "| #{"-" * name_rjust} | #{"-" * time_ljust} | #{"-" * mem_rjust} |"
+  total.each do |name, t|
+    print "| "
+    print name.rjust(name_rjust)
+    print " | "
+    print "#{t[:time]}ms".ljust(time_ljust)
+    print " | "
+    print t[:memory].humanize_bytes.ljust(mem_rjust)
+    puts " |"
+  end
+  puts "\nTotal Time  : #{total.sum { |_, v| v[:time] }}ms"
+  puts "Total Memory: #{total.sum { |_, v| v[:memory] }.humanize_bytes}"
+end
+
+macro benchmark(&)
+  image = Pluto::Image.from_ppm(File.read("lib/pluto_samples/pluto.ppm"))
+  memory = 0i64
+  time = benchmark_time do
+    memory = benchmark_memory do
+      {{yield}}
+    end
+  end
+  total["{{yield.id}}".gsub(/^image\./, "")] = {memory: memory, time: time.total_milliseconds.to_i64}
```

Comment:
Can we avoid the regex by using `"\nimage."` instead?

Reply:
Any particular reason to avoid the regex? The string replacement with the newline as first char doesn't work. I can do it with only `"image."` instead, which feels like a safe assumption.

Reply:
I think using regex when there's no reason to is a bad practice in general. In this case `"image."` makes code easier to read.